src/Kunstmaan/NodeSearchBundle/Services/SearchService.php

```diff
@@ -120,13 +125,28 @@ public function search()
         $request = $this->requestStack->getCurrentRequest();
 
         // Retrieve the current page number from the URL, if not present of lower than 1, set it to 1
+        $pageNumber = $this->getRequestedPage($request);
         $entity = $request->attributes->get('_entity');
 
-        $pageNumber = $this->getRequestedPage($request);
-        $searcher   = $this->container->get($entity->getSearcher());
+        if (!empty($this->searchers)) {
+            foreach ($this->searchers as $possibleSearcher) {
+                if ($entity->getSearcher() === \get_class($possibleSearcher)) {
+                    $searcher = $possibleSearcher;
+                }
+            }
+        }
+
+        if (!isset($searcher)) {
+            @trigger_error(
+                'Getting the node searcher from the container is deprecated in KunstmaanNodeSearchBundle 5.1 and will be removed in KunstmaanNodeSearchBundle 6.0.
```

Comment:
Maybe use `Getting the node searcher "*the actual service id*" ...` so the users directly know which service to tag